tracee/printer.go

```diff
@@ -167,6 +180,42 @@ func (p tableEventPrinter) Epilogue(stats statsStore) {
 	fmt.Fprintf(p.out, "Stats: %+v\n", stats)
 }
 
+
+type templateEventPrinter struct {
+	tracee        *Tracee
+	out           io.Writer
+	err           io.Writer
+	containerMode bool
+	templatePath  string
+}
+
+func (p templateEventPrinter) Init() {}
+
+func (p templateEventPrinter) Preamble() {}
+
+func (p templateEventPrinter) Error(err error) {
+	fmt.Fprintf(p.err, "%v", err)
+}
+
+
+func (p templateEventPrinter) Print(event Event) {
+	tmplPath := p.templatePath
+	if tmplPath != "" {
+		tmpl, err := template.ParseFiles(tmplPath)
+		if err != nil {
+			p.Error(err)
+		}
+		err = tmpl.Execute(p.out, event)
+		if err != nil {
+			p.Error(err)
+		}
+	} else {
+		p.Error(errors.New("Please specify a go-template for event-based output"))
```

Comment:
We'll need some sort of basic validation to occur earlier. I think that a good place would be in the Printer's `Init` phase. For that, you can modify the signature of `Init` to return an error.

Reply:
Yeah the Init has been modified now, could you please have a look?

Reply:
What I meant here is that Init should validate the input, and fail if it's invalid. In your implementation it doesn't fail it just prints that there was an error for every event, but the error is known in init time so there no reason to continue trying.
To be more specific, you need to change the Init function (and interface definition) to return an error, and bubble that error up until it fails the program (which is a good thing if the input is invalid)

Reply:
Ohh okk got it...I'll make the change then.....do I have to modify the Init signatures of other templates as well?

Reply:
You would need to change them as well just for the program to compile but you don't need to implement any logic there.

Reply:
@itaysk I've made the change.....could you please have a look?

Reply:
You have changed the signature of Init, but it is never called. We discussed you will add it to the interface of `eventPrinter` so that the calling code can initialize the eventPrinter.